cmd/node/main.go

```diff
@@ -1332,10 +1332,17 @@ func startNode(ctx *cli.Context, log logger.Logger, version string) error {
 		log.Info("terminating at internal stop signal", "reason", sig.Reason)
 	}
 
+	chanCloseComponents := make(chan struct{})
 	go func() {
-		closeAllComponents(log, dataComponents, triesComponents, networkComponents)
+		closeAllComponents(log, dataComponents, triesComponents, networkComponents, chanCloseComponents)
 	}()
-	time.Sleep(maxTimeToClose)
+
+	select {
+	case <-chanCloseComponents:
+	case <-time.After(maxTimeToClose):
+		log.Debug("closeAllComponents did not finished on time, force closing the node")
```

Comment:
log.Warn / error maybe ?

Reply:
add error in the message in order to find it easily.

Reply:
done